### DISCUSSION TITLE
Process ID for Qutebrowser

### DISCUSSION THREAD

**Discussion by wgmayer0** [qutebrowser/qutebrowser#6494] [Q&A] [answered]
Does Qutebrowser have a unique process ID on linux?

My system: Arch Linux
Name            : qutebrowser
Version         : 2.2.2-1

I tried to find out if it did by typing `xprop` and then clicking this exact window that I am typing this on. The output was:

`_NET_WM_PID(CARDINAL) = 16583`
 
I the tried the same thing on a new instance of Qutebrowser. I went to msn.com then typed `xprop` again and then clicked the new instance. After doing this, I get the exact same process ID:

`_NET_WM_PID(CARDINAL) = 16583`

Is there no such thing as a unique number that identifies each individual instance of Qutebrowser? I want this so that I can write bash scripts that specifically apply to individual windows.

**Comment by The-Compiler** [comment, marked as answer]
Like pretty much any other browser, multiple windows will run in the same instance/process. So there is only one instance - I assume you mean something that identifies a single *window* in that instance?

You could add `{win_id}` to `window.title_format`, or perhaps start a new isolated instance using a different `--basedir` or `--temp-basedir`. Finally, note that windows have an X11 window ID as well, I guess it really depends on what you're writing exactly.
